### ISSUE TITLE
LMOD_CACHED_LOADS breaks module loading in lmod 7.1

### ISSUE THREAD

**Issue by fretn** [TACC/Lmod#204]
```
[user@host ~]$ export LMOD_CACHED_LOADS=no
[user@host ~]$ module purge
[user@host ~]$ module load blast
[user@host ~]$ module load perl
[user@host ~]$ export LMOD_CACHED_LOADS=1
[user@host ~]$ module purge
[user@host ~]$ module load blast
Lmod has detected the following error:  Unable to load module: blast
     /software/shared/lmodulefiles/general/blast/2.5.0+ : ...are/shared/apps/x86_64/lmod/7.1/libexec/FrameStk.lua:127: attempt to index local 'mname' (a boolean value)




[user@host ~]$ module load perl
Lmod has detected the following error:  These module(s) exist but cannot be loaded as requested: "perl"

   Try: "module spider perl" to see how to load the module(s).





[user@host ~]$
```

when LMOD_CACHED_LOADS is set to 1 then module load doesn't work at all, I get two different errors .
Is this a known problem or have I configured something wrong ?

**Comment by rtmclay** [comment]
There were some problem in Lmod 7+ w.r.t LMOD_CACHED_LOADS=1.   You can see the changes to Lmod in README.new.   I use this file record changes between Lmod X.Y and Lmod X.Y+1 (currently changes since 7.1). There was a fix in 7.1.1 and 7.1.3 to fix problems with LMOD_CACHED_LOADS=1.  

You should try Lmod 7.1.6 to see if it fixes your problem.

**Comment by fretn** [comment]
my bad, didn't check if there where any updates since 7.1
